### PR TITLE
Add migration doc 0.3 to 0.4

### LINKED ISSUE #421: Migration doc for going from 0.3 to 0.4
This content should live in the changelog for 0.4.

Relates to #378

### PR DESCRIPTION
**Pull request by Weakky**
[Rendered view](https://github.com/prisma-labs/nexus-prisma/blob/chore/migration-doc/0.3-to-0.4.md)

Closes #421

### CODE REVIEW

File: 0.3-to-0.4.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Migrating from 0.3.x to 0.4.x
+
+This document helps you migrate from `nexus-prisma@0.3.x` to `nexus-prisma@0.4.x`.
+
+⚠️  **`nexus-prisma@0.4.x` is still in preview and might break at any moment. We advise not to upgrade yet if you're running in production.**
```
Comment (jasonkuhrt) [inline 0.3-to-0.4.md:5]: How was `0.3.x` production ready though? >>> `0.3` <<< 😂

**Review by jasonkuhrt — APPROVED** [review]
Instead of putting this into the repo, how about just inlining this markdown into the release notes? Avoids having to host the file in the repo. And I think migration guides are generally something we'll want to keep in release notes (future ones too).

File: 0.3-to-0.4.md (hunk 1 above)
Comment (Weakky) [reply 0.3-to-0.4.md:5]: Well, even though it wasn't, lots of people used it in prod 🙈

**Comment by Weakky** [comment]
Updated the doc to fit in the release note & updated the release note. Closing now

**Comment by jasonkuhrt** [comment]
@Weakky fyi made some formatting tweaks to https://github.com/prisma-labs/nexus-prisma/releases/tag/v0.4.1 as it gets a bit harder to read the headings in the release context. Namely using bullet points, and moving the small migration guide (pre-release) to top.

Great work I think this is much better than before 🙌

**Pull request closed without merging**